src/main/java/de/tum/in/www1/artemis/service/MailService.java

```diff
@@ -190,7 +190,8 @@ else if (notificationSubject instanceof FileUploadExercise) {
                 context.setVariable(EXERCISE_TYPE, "upload");
             }
         }
-        else if (notificationSubject instanceof Post) {
+
+        if (notificationSubject instanceof Post) {
```

Comment:
Is there a particular reason why you removed the `else if` and replaced it with `if`?

Reply:
Thank you very much for your review Olimbia, yes (I think Lorena and I accidentally added a link breaking change in [#4142](https://github.com/ls1intum/Artemis/pull/4142))

And once again fixed it in [#4303](https://github.com/ls1intum/Artemis/pull/4303)
![image](https://user-images.githubusercontent.com/65814168/142736767-dfb46978-0552-439b-9e5d-be87dc6a254a.png)
I just fixed it here also but I assume that Lorena's PR will be merged first so yeah. We will see^^

Reply:
With the else the URL is not set correctly for every case besides POSTs ;/

Reply:
Thank you for explaining it @Malyuk-A